crates/partition-store/src/outbox_table/mod.rs

```diff
@@ -82,14 +82,35 @@ fn get_outbox_message<S: StorageAccess>(
 fn truncate_outbox<S: StorageAccess>(
     storage: &mut S,
     partition_id: PartitionId,
-    seq_to_truncate: Range<u64>,
+    range: RangeInclusive<u64>,
 ) {
-    let mut key = OutboxKey::default().partition_id(partition_id);
-    let k = &mut key;
+    let _x = RocksDbPerfGuard::new("truncate-outbox");
+
+    let mut keys = vec![];
+    {
+        let start_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(*range.start());
+
+        let end_key = OutboxKey::default()
+            .partition_id(partition_id)
+            .message_index(*range.end());
+
+        let mut iter = storage.iterator_from(TableScan::KeyRangeInclusiveInSinglePartition(
+            partition_id,
+            start_key,
+            end_key,
+        ));
+
+        iter.seek_to_last();
+        while iter.valid() {
+            keys.push(iter.key().unwrap().to_vec());
+            iter.prev();
+        }
+    }
```

Comment:
Not at all important nor intended. I adopted the scan-and-delete (anti)pattern because 1) I didn't find the `delete_range_cf` operation on `StorageAccess` (nee `rocksdb::DBAccess`), but also 2) we don't (yet) track of the lower bound of unpruned outbox sequence ids.

Reply:
Regarding 2): The API expects a range and we are limiting the scan to this range. So why not directly deleting every entry between this range. Whether the range is correctly set or not, is then an orthogonal problem.

Reply:
That's absolutely viable! I was being overly paranoid about making a counted loop over a large range initially but given this is likely always going to be on the order of a range of 1-2 sequence ids, it's absolutely unnecessary. Let me revise that - with this optimisation in place, the PR as a whole should be viable to merge.

Reply:
Now tracking low/high sequence ids explicitly and truncating between the two.